examples/usdt_sample/scripts/bpf_text_shared.c

```diff
@@ -4,15 +4,28 @@
 /**
  * @brief Helper method to filter based on the specified inputString.
  * @param inputString The operation input string to check against the filter.
- * @return True if the specified inputString starts with the hard-coded FILTER_STRING; otherwise, false.
+ * @return True if the specified inputString starts with the hard-coded filter string; otherwise, false.
  */
 static inline bool filter(char const* inputString)
 {
-    char needle[] = "FILTER_STRING"; ///< The FILTER STRING is replaced by python code.
-    char haystack[sizeof(needle)] = {};
-    bpf_probe_read_user(&haystack, sizeof(haystack), (void*)inputString);
-    for (int i = 0; i < sizeof(needle) - 1; ++i) {
-        if (needle[i] != haystack[i]) {
+    static const char* null_ptr = 0x0;
+    static const char null_terminator = '\0';
+
+    static const char filter_string[] = "FILTER_STRING"; ///< The filter string is replaced by python code.
+    if (null_ptr == inputString) {
+        return false;
+    }
+    // bpf_trace_printk("inputString: '%s'", inputString);
```

Comment:
could you remove the commented-out printk?

Reply:
Done